src/spider/worker/worker.cpp

```diff
@@ -369,40 +365,86 @@ auto task_loop(
-            case spider::core::TaskLanguage::Python:
-                break;
-            default:
+        std::unique_ptr<spider::worker::TaskExecutor> executor = nullptr;
+        // Task setup scope to ensure storage connection RAII
+        {
+            std::variant<std::unique_ptr<spider::core::StorageConnection>, spider::core::StorageErr>
+                    conn_result = storage_factory->provide_storage_connection();
+            if (std::holds_alternative<spider::core::StorageErr>(conn_result)) {
+                spdlog::error(
+                        "Failed to connect to storage: {}",
+                        std::get<spider::core::StorageErr>(conn_result).description
+                );
+                continue;
+            }
+            auto conn = std::move(
+                    std::get<std::unique_ptr<spider::core::StorageConnection>>(conn_result)
+            );
+
+            std::optional<std::vector<msgpack::sbuffer>> optional_arg_buffers
+                    = setup_task(*conn, metadata_store, instance, task);
+            if (!optional_arg_buffers.has_value()) {
+                spdlog::error("Failed to setup task {}", task.get_function_name());
+                fail_task_id = task.get_id();
+                continue;
+            }
+            std::vector<msgpack::sbuffer> const& arg_buffers = optional_arg_buffers.value();
+
+            // Validate task language
+            auto const language = task.get_language();
+
+            // Execute task
+            if (spider::core::TaskLanguage::Cpp == language) {
+                executor = spider::worker::TaskExecutor::spawn_cpp_executor(
+                        context,
+                        task.get_function_name(),
+                        task.get_id(),
+                        storage_url,
+                        libs,
+                        environment,
+                        arg_buffers
+                );
+            } else if (spider::core::TaskLanguage::Python == language) {
+                executor = spider::worker::TaskExecutor::spawn_python_executor(
+                        context,
+                        task.get_function_name(),
+                        task.get_id(),
+                        storage_url,
+                        environment,
+                        arg_buffers
+                );
+            } else {
                 spdlog::error("Unsupported task language.");
                 fail_task_id = task.get_id();
+                metadata_store->task_fail(
+                        *conn,
+                        instance,
+                        fmt::format(
+                                "Unsupported task language for task {}",
+                                task.get_function_name()
+                        )
+                );
+                continue;
+            }
+            if (nullptr == executor) {
+                spdlog::error(
+                        "Failed to spawn task executor for task {}",
```

Comment:
```suggestion
                        "Failed to spawn task executor for task `{}`.",
```
Period.
Fix all log statements to be consistent with line 417.